app/models/post.rb

```diff
@@ -12,6 +12,15 @@ class Post < ApplicationRecord
 
   def publish!
     self.status = :published
+    self.published_at = Time.now
     save
   end
+
+  def path
+    "post/#{self.slug}"
+  end
```

Comment:
I didn't get this method's purpose. The friendly_id gem is supposed to take care of this for us